src/Mapping/MappedEventSubscriber.php

```diff
@@ -291,7 +292,12 @@ private function getCacheItemPool(ObjectManager $objectManager): CacheItemPoolIn
         }
 
         if ($objectManager instanceof EntityManagerInterface || $objectManager instanceof DocumentManager) {
-            $metadataCache = $objectManager->getConfiguration()->getMetadataCache();
+            $metadataFactory = $objectManager->getMetadataFactory();
+            $getCache = \Closure::bind(static function (AbstractClassMetadataFactory $metadataFactory): ?CacheItemPoolInterface {
```

Comment:
At first sight, violating the encapsulation seems a workaround.
Do we have any chance to add a TODO explaining how we should resolve this problem in a proper way?

Reply:
I've added a comment to try to explain it, for the CS fail I've created https://github.com/doctrine-extensions/DoctrineExtensions/pull/2488

Reply:
Thank you! #2488 is merged. Let's rebase this :rocket:

Reply:
🚀🚀